cts-java/src/test/java/org/ga4gh/cts/api/references/ReferencesPagingIT.java

```diff
@@ -0,0 +1,149 @@
+package org.ga4gh.cts.api.references;
+
+import org.apache.avro.AvroRemoteException;
+import org.ga4gh.ctk.transport.URLMAPPING;
+import org.ga4gh.ctk.transport.protocols.Client;
+import org.ga4gh.cts.api.Utils;
+import org.ga4gh.cts.api.datasets.DatasetsTests;
+import org.ga4gh.methods.GAException;
+import org.ga4gh.methods.SearchReferencesRequest;
+import org.ga4gh.methods.SearchReferencesResponse;
+import org.ga4gh.models.Reference;
+import org.ga4gh.models.ReferenceSet;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Test paging through the results returned by <tt>/referencesets/search</tt>.
+ *
+ * @author Herb Jellinek
+ */
+@Category(DatasetsTests.class)
+public class ReferencesPagingIT {
+
+    private static Client client = new Client(URLMAPPING.getInstance());
+
+    /**
+     * Check that we can page 1 by 1 through the {@link org.ga4gh.models.Reference}s we receive from
+     * {@link org.ga4gh.ctk.transport.protocols.Client.References#searchReferences(SearchReferencesRequest)}.
+     *
+     * @throws AvroRemoteException if there's a communication problem or server exception ({@link GAException})
+     */
+    @Test
+    public void checkPagingOneByOneThroughReferences() throws AvroRemoteException {
+
+        final List<ReferenceSet> allRefSets = Utils.getAllReferenceSets(client);
+        assertThat(allRefSets).isNotEmpty();
+
+        final String refSetId = allRefSets.get(0).getId();
+
+        // retrieve them all
+        final List<Reference> listOfReferences = Utils.getAllReferences(client, refSetId);
+        assertThat(listOfReferences).isNotEmpty();
+
+        // we will remove References from this Set and assert at the end that we have zero
+        final Set<Reference> setOfReferences = new HashSet<>(listOfReferences);
+        assertThat(listOfReferences).hasSize(setOfReferences.size());
+
+        // page through the References using the same query parameters
+        String pageToken = null;
+        for (Reference ignored : listOfReferences) {
```

Comment:
This seems quite low-level. Why not provide a method to do the paging automatically, which you can provide an optional page size (which I think @calbach suggests above too)? There's going to be an awful lot of code repetition if this is done for every object, on every search. I would say this test should be replaced with something like
```java
final Set<Reference> expectedReferences = Set(Utils.getAllReferences(client, refSetId));
final Set<Reference> foundReferences = Set(getReferences(client, refSetId, pageSize=1));
assertThat(expectedReferences == foundReferences);
```
(apologies for the mangled pseudocode, but you get the idea!)

The point is, we can't really every assert anything interesting about pageSizes other than they are > 0, so there's no point in checking these things.